### PR TITLE
Refactors pipe.dm

### PR DESCRIPTION
**Pull request by Laboredih123**
[ATMOSPHERICS][INTERNAL][CLEAN]
## About the PR 
Refactors pipe.dm into being cleaner, moves out some types that should eventually be changed, merges regular_pipe.dmi and color_pipe.dmi so we don't have to deal with two image files, and some more.

## Why's this needed? 
It's about time pipes were refactored.

### CODE REVIEW

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+/obj/machinery/atmospherics/pipe/tank/New()
+	..()
+	initialize_directions = dir
+
+/obj/machinery/atmospherics/pipe/tank/process()
+	..()
+	if(!node1)
+		parent.mingle_with_turf(loc, 200)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:20]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	name = "Pressure Tank (Carbon Dioxide)"
+
+/obj/machinery/atmospherics/pipe/tank/carbon_dioxide/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.carbon_dioxide = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:31]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	name = "Pressure Tank (Plasma)"
+
+/obj/machinery/atmospherics/pipe/tank/toxins/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.toxins = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:43]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	name = "Pressure Tank (Oxygen + Plasma)"
+
+/obj/machinery/atmospherics/pipe/tank/oxygen_agent_b/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T0C
+
+	air_temporary.oxygen_agent_b = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:55]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+	name = "Pressure Tank (Oxygen)"
+
+/obj/machinery/atmospherics/pipe/tank/oxygen/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.oxygen = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:67]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+	name = "Pressure Tank (Nitrogen)"
+
+/obj/machinery/atmospherics/pipe/tank/nitrogen/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.nitrogen = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:79]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+	name = "Pressure Tank (N2O)"
+
+/obj/machinery/atmospherics/pipe/tank/sleeping_agent/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.nitrous_oxide = (50*ONE_ATMOSPHERE)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:91]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+/obj/machinery/atmospherics/pipe/tank/air/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.oxygen = (50*ONE_ATMOSPHERE*O2STANDARD)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+	air_temporary.nitrogen = (50*ONE_ATMOSPHERE*N2STANDARD)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:104]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+/obj/machinery/atmospherics/pipe/tank/air_repressurization/New()
+	..()
+	air_temporary = new /datum/gas_mixture
+	air_temporary.volume = volume
+	air_temporary.temperature = T20C
+
+	air_temporary.oxygen = (180*ONE_ATMOSPHERE*O2STANDARD)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+	air_temporary.nitrogen = (180*ONE_ATMOSPHERE*N2STANDARD)*(air_temporary.volume)/(R_IDEAL_GAS_EQUATION*air_temporary.temperature)
+
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:121]:
```suggestion


```

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+/obj/machinery/atmospherics/pipe/tank/disposing()
+	node1?.disconnect(src)
+	parent = null
+	..()
+
+/obj/machinery/atmospherics/pipe/tank/pipeline_expansion()
+	return list(node1)
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/unary/pressure_tank.dm:128]: all of this should probably go above the subtypes for this case

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/unary/pressure_tank.dm (hunk 10 above)
Comment (Laboredih123) [reply code/modules/atmospherics/machinery/unary/pressure_tank.dm:128]: ight

**Comment by Laboredih123** [comment]
yay almost done with refactoring

**Pull request merged**